x-pack/solutions/security/plugins/security_solution/public/asset_inventory/pages/all_assets.tsx

```diff
@@ -0,0 +1,431 @@
+
+  const [columns, setColumns] = useLocalStorage(
+    columnsLocalStorageKey,
+    defaultColumns.map((c) => c.id)
+  );
+
+  const [persistedSettings, setPersistedSettings] = useLocalStorage<UnifiedDataTableSettings>(
+    `${columnsLocalStorageKey}:settings`,
+    {
+      columns: defaultColumns.reduce((columnSettings, column) => {
+        const columnDefaultSettings = column.width ? { width: column.width } : {};
+        const newColumn = { [column.id]: columnDefaultSettings };
+        return { ...columnSettings, ...newColumn };
+      }, {} as UnifiedDataTableSettings['columns']),
+    }
+  );
+
+  const settings = useMemo(() => {
+    return {
+      columns: Object.keys(persistedSettings?.columns as UnifiedDataTableSettings).reduce(
+        (columnSettings, columnId) => {
+          const newColumn: UnifiedDataTableSettingsColumn = {
+            ..._.pick(persistedSettings?.columns?.[columnId], ['width']),
+            display: columnHeaders?.[columnId],
+          };
+
+          return { ...columnSettings, [columnId]: newColumn };
+        },
+        {} as UnifiedDataTableSettings['columns']
+      ),
+    };
+  }, [persistedSettings]);
+
+  const { dataView, dataViewIsLoading, dataViewIsRefetching } = useDataViewContext();
+
+  const [expandedDoc, setExpandedDoc] = useState<DataTableRecord | undefined>(undefined);
+
+  const renderDocumentView = (hit: DataTableRecord) =>
+    flyoutComponent(hit, () => setExpandedDoc(undefined));
+
+  const {
+    uiActions,
+    uiSettings,
+    dataViews,
+    data,
+    application,
+    theme,
+    fieldFormats,
+    notifications,
+    storage,
+  } = useKibana().services;
+
+  const styles = useStyles();
+
+  const { capabilities } = application;
+  const { filterManager } = data.query;
+
+  const services = {
+    theme,
+    fieldFormats,
+    uiSettings,
+    toastNotifications: notifications.toasts,
+    storage,
+    data,
+  };
+
+  const {
+    columns: currentColumns,
+    onSetColumns,
+    onAddColumn,
+    onRemoveColumn,
+  } = useColumns({
+    capabilities,
+    defaultOrder: uiSettings.get(SORT_DEFAULT_ORDER_SETTING),
+    dataView,
+    dataViews,
+    setAppState: (props) => setColumns(props.columns),
+    columns,
+    sort,
+  });
+
+  /**
+   * This object is used to determine if the table rendering will be virtualized and the virtualization wrapper height.
+   * mode should be passed as a key to the UnifiedDataTable component to force a re-render when the mode changes.
+   */
+  const computeDataTableRendering = useMemo(() => {
+    // Enable virtualization mode when the table is set to a large page size.
+    const isVirtualizationEnabled = pageSize >= 100;
+
+    const getWrapperHeight = () => {
+      if (height) return height;
+
+      // If virtualization is not needed the table will render unconstrained.
+      if (!isVirtualizationEnabled) return 'auto';
+
+      const baseHeight = 362; // height of Kibana Header + Findings page header and search bar
```

Comment:
that's likely to be different for the asset inventory page, but we can think of a more generic solution once we consolidate the datagrid as a package.